src/mds/Server.cc

```diff
@@ -5722,6 +5722,28 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::return_bytes(string& quota_str)
+{
+  unordered_map<char, int64_t> size_value = {{'k', 1L << 10}, {'m', 1L << 20}, 
+                                             {'g', 1L << 30}, {'t', 1L << 40}};
+  char last_char = tolower(quota_str.back());
+  if (size_value.find(last_char) != size_value.end()) {
+    try {
+      return (boost::lexical_cast<int64_t>(quota_str.substr(0, quota_str.length() - 1))) * size_value[last_char];
+    } catch (boost::bad_lexical_cast const&) {
+      dout(10) << "invalid numberical value encountered" << dendl;
```

Comment:
if you are logging the issue here, you might as well dump the offending value ?

Reply:
yup, makes sense. will add it.

Reply:
done. PTAL